commit d2019da3b897665515e7ea3c2a2ab9e8f2b3beb7
Author: Jan Kara <jack@suse.cz>
Date:   Fri Jan 27 14:34:30 2017 -0500

    ext4: trim allocation requests to group size
    
    commit cd648b8a8fd5071d232242d5ee7ee3c0815776af upstream.
    
    If filesystem groups are artifically small (using parameter -g to
    mkfs.ext4), ext4_mb_normalize_request() can result in a request that is
    larger than a block group. Trim the request size to not confuse
    allocation code.
    
    Reported-by: "Kirill A. Shutemov" <kirill.shutemov@linux.intel.com>
    Signed-off-by: Jan Kara <jack@suse.cz>
    Signed-off-by: Theodore Ts'o <tytso@mit.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linux-4.9.47/fs/ext4/mballoc.c
===================================================================
--- linux-4.9.47.orig/fs/ext4/mballoc.c
+++ linux-4.9.47/fs/ext4/mballoc.c
@@ -3130,6 +3130,13 @@ ext4_mb_normalize_request(struct ext4_al
 	if (size > EXT4_BLOCKS_PER_GROUP(ac->ac_sb))
 		size = EXT4_BLOCKS_PER_GROUP(ac->ac_sb);
 
+	/*
+	 * Trim allocation request for filesystems with artificially small
+	 * groups.
+	 */
+	if (size > EXT4_BLOCKS_PER_GROUP(ac->ac_sb))
+		size = EXT4_BLOCKS_PER_GROUP(ac->ac_sb);
+
 	end = start + size;
 
 	/* check we don't cross already preallocated blocks */
Index: linux-4.9.47/dummy/rpi_1278_d2019da3b897665515e7ea3c2a2ab9e8f2b3beb7.txt
===================================================================
--- /dev/null
+++ linux-4.9.47/dummy/rpi_1278_d2019da3b897665515e7ea3c2a2ab9e8f2b3beb7.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
